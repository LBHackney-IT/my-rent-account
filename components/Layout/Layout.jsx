import React from "react";
import PropTypes from "prop-types";

import Header from "./Header/Header";
import Footer from "./Footer/Footer";
import SkipLink from "./SkipLink/SkipLink";
import PhaseBanner from "./PhaseBanner/PhaseBanner";

const Layout = ({ children }) => (
  <>
    <SkipLink />
    <Header serviceName="My rent account" />
    <div className="govuk-width-container app-width-container">
      <PhaseBanner phase="beta" />
      <main
        className="govuk-main-wrapper app-main-class"
        id="content"
        role="main"
      >
        {children}
      </main>
    </div>
    <Footer />
  </>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
