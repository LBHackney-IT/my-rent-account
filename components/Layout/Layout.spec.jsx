import React from "react";
import { render } from "@testing-library/react";

import Layout from "./Layout";

jest.mock("lbh-frontend-react", () => ({
  Container: "lbh-Container",
  Header: "lbh-Header",
  Main: "lbh-Main",
  Paragraph: "lbh-Paragraph",
  Tag: "lbh-Tag",
}));

describe("Layout component", () => {
  it("should render properly", () => {
    const { asFragment } = render(
      <Layout>
        <div>foo</div>
      </Layout>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
