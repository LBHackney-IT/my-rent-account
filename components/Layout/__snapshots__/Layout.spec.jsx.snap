// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render properly 1`] = `
<DocumentFragment>
  <a
    class="govuk-skip-link"
    href="#content"
  >
    Skip to main content
  </a>
  <header
    servicename="My rent account"
  />
  <div
    class="govuk-width-container app-width-container"
  >
    <div
      class="govuk-phase-banner"
    >
      <p
        class="govuk-phase-banner__content"
      >
        <strong
          class="govuk-tag govuk-phase-banner__content__tag"
        >
          beta
        </strong>
        <span
          class="govuk-phase-banner__text"
        >
          This is a new service â€“ your 
          <a
            class="govuk-link"
          >
            feedback
          </a>
           will help us to improve it.
        </span>
      </p>
    </div>
    <main
      class="govuk-main-wrapper app-main-class"
      id="content"
      role="main"
    >
      <div>
        foo
      </div>
    </main>
  </div>
  <footer />
</DocumentFragment>
`;
